controllers/rcj_soccer_referee_supervisor/referee/penalty_area_checker.py

```diff
@@ -0,0 +1,73 @@
+from typing import List
+
+from referee.consts import BLUE_PENALTY_AREA, YELLOW_PENALTY_AREA
+
+
+class PenaltyAreaChecker:
+    def __init__(self, allowed: int, reset_after: int, time: int):
+        self.allowed = allowed
+        self.reset_after = reset_after
+        self.time = time
+
+        self.y_vertical, self.y_lower, self.y_upper = YELLOW_PENALTY_AREA
+        self.b_vertical, self.b_lower, self.b_upper = BLUE_PENALTY_AREA
+        self.reset()
+
+    def reset(self):
+        self.time_entered_penalty = -1
+        self.time_left_penalty = -1
+
+    def is_in_yellow_penalty(self, x: int, z: int) -> bool:
+        return x < self.y_vertical and self.y_lower < z < self.y_upper
+
+    def is_in_blue_penalty(self, x: int, z: int) -> bool:
+        return x > self.b_vertical and self.b_lower < z < self.b_upper
+
+    @property
+    def is_outside_of_penalty_for_longer(self) -> bool:
+        return self.time < self.time_left_penalty - self.reset_after
+
+    @property
+    def is_inside_of_penalty_for_long(self) -> bool:
+        return self.time < self.time_entered_penalty - self.allowed
+
+    @property
+    def has_entered(self) -> bool:
+        return self.time_entered_penalty != -1
```

Comment:
@Adman just a quick thought: do you think extracting `-1` to some constant would make sense here? Probably not, but I still wanted to ask.

Reply:
I think we can change `-1` to `None` instead.